File: Progammers/연습문제/Level 1/소수 만들기/Solution.java
import java.util.Arrays;

class Solution {
    public int solution(int[] nums) {
        int ans = 0;

        for(int i = 0; i < nums.length - 2; i ++){
            for(int j = i + 1; j < nums.length - 1; j ++){
                for(int k = j + 1; k < nums.length; k ++ ){
                    if(isPrime(nums[i] + nums[j] + nums[k])){
                        ans += 1;  
                    } 
                }
            }
        }
        return ans;
    }
    public boolean isPrime(int target){
        boolean flag = true;
        for(int d=2; d<=(int)Math.sqrt(target); d++){
            if(target % d == 0){
                flag = false;
                break;
            }
        }
        return flag;
    }
}

File: Progammers/연습문제/Level 1/소수 만들기/Solution.js
function solution(nums) {
    const answer = sumOfThreeIsPrime(nums);
    return answer;
}

function sumOfThreeIsPrime(arr) {
    let result = 0;
    for (let i=0; i<arr.length; i++) {
        for (let j=i+1; j<arr.length; j++) {
            for (let k=j+1; k<arr.length; k++) {
                if (isPrime(arr[i]+arr[j]+arr[k])) {
                    result++;
                }
            }
        }
    }
    return result;
}

function isPrime(number) {
    for (let div=2; div<Math.sqrt(number)+1; div++) {
        if (number % div == 0) {
            return false;
        }
    }
    return true;
}


// TC
const nums = [1, 2, 3, 4];
console.log(solution(nums)); // 1